### PR TITLE
OAK-9984: FlatFileSplitterTest reliably failing

### PR DESCRIPTION
**Pull request by amit-jain**
- Replace file length comparisons with file content comparison

### CODE REVIEW

File: oak-run-commons/src/test/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/FlatFileSplitterTest.java [hunk 1]
```diff
@@ -533,4 +548,15 @@ public void uncompress(File src, File dest) throws IOException {
             }
         }
     }
+    
+    private static Set<String> readAllFilesAsSet(List<File> files) {
+        Set<String> set = files.stream().flatMap(file -> {
+            try ( InputStream is = Files.newInputStream(file.toPath())) {
```
Comment (reschke) [inline oak-run-commons/src/test/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/FlatFileSplitterTest.java]:
```suggestion
            try (InputStream is = Files.newInputStream(file.toPath())) {
```

**Review by reschke — APPROVED** [review]
This indeed seems to eliminate test failures.

I still see leftover files in TMPDIR, but we can handle that as a less pressing, separate issue.

File: oak-run-commons/src/test/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/FlatFileSplitterTest.java (hunk 1 above)
Comment (amit-jain) [reply oak-run-commons/src/test/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/FlatFileSplitterTest.java]: > I still see leftover files in TMPDIR, but we can handle that as a less pressing, separate issue.

This should be fixed with my latest commits to
- Creating TemporaryFolder into the target directory
- Setting the system property java.tmpdir under the temporary folder

Comment (reschke) [reply oak-run-commons/src/test/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/FlatFileSplitterTest.java]: Hm. My main concern is not leftover files from tests, but unclosed files in general. These may not manifest themselves as leftover files on non-Windows systems, but would still be a problem in production.

Comment (amit-jain) [reply oak-run-commons/src/test/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/FlatFileSplitterTest.java]: Yes good point. I am checking that too.

**Pull request merged**